frontend/src/hocs/withHostCommunication/withHostCommunication.tsx

```diff
@@ -97,7 +144,7 @@ function withHostCommunication(
         }
 
         if (message.type === "SET_AUTH_TOKEN") {
-          setAuthToken(message.authToken)
+          deferredAuthToken.resolve(message.authToken)
```

Comment:
In this case the promise resolution is actually a no-op since the promise would have already been resolved to `undefined` earlier. I think it's probably not worth explicitly handling this case since it adds complexity for an edge case that should never occur, and it doesn't actually change any behavior if we do more explicitly handle the scenario.

I'll add a comment explaining why we're covered in this situation, though